mlflow/langchain/__init__.py

```diff
@@ -800,10 +801,16 @@ def load_model(model_uri, dst_path=None):
     return _load_model_from_local_fs(local_model_path)
 
 
-def _load_code_model(code_path: Optional[str] = None):
+@contextmanager
+def _config_path_context(code_path: Optional[str] = None):
     _set_config_path(code_path)
+    yield
+    _set_config_path(None)
+
 
-    import chain  # noqa: F401
+def _load_code_model(code_path: Optional[str] = None):
+    with _config_path_context(code_path):
+        import chain  # noqa: F401
```

Comment:
the `noqa` because this is set as a module field? Should we wrap this up in a try..except?

Reply:
Its a lint failure that we are importing a module before never using it.